component/discovery/kubernetes_crds/kubernetes_crds.go

```diff
@@ -0,0 +1,336 @@
+package kubernetes_crds
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/prometheus"
+	"github.com/grafana/agent/pkg/util/k8sfs"
+	promop "github.com/prometheus-operator/prometheus-operator/pkg/client/informers/externalversions"
+	promCommonConfig "github.com/prometheus/common/config"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/prometheus/prometheus/scrape"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus-operator/prometheus-operator/pkg/client/versioned"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "prometheus.crds",
```

Comment:
Is the intent that this one component handles all the Prometheus CRD types? I need to think about that one a little, I'm not sure if that's going to complicate the debug information or the configuration at all, and it turns this component into a bit of a monolith.